pkg/controller/migration/convert/cni.go

```diff
@@ -42,11 +75,30 @@ func loadCNI(c *components) error {
 			if err := json.Unmarshal(plugin.Bytes, &c.calicoCNIConfig); err != nil {
 				return fmt.Errorf("failed to parse calico cni config: %w", err)
 			}
+			if plugin.Network.IPAM.Type == "host-local" {
+				// First load the IPAM json section so we can Unmarshal it below.
+				// We load the IPAM as raw json so we can isolate it from the rest of the
+				// CNI config.
+				raw := CNIHostLocalIPAM{}
+				if err := json.Unmarshal(plugin.Bytes, &raw); err != nil {
+					return fmt.Errorf("failed to parse cni config for raw IPAM: %w", err)
+				}
+
+				// Use a Decoder so we can set DisallowUnknownFields so if there are
+				// any unknown fields we will detect that.
+				x := HostLocalIPAMConfig{}
+				dec := json.NewDecoder(bytes.NewReader(raw.IPAM))
+				dec.DisallowUnknownFields()
```

Comment:
Similar to my previous comment, I wonder if at a higher level we don't need to go validating that their host-local CNI is valid, and can just parse the bits we need...

Reply:
My intention here wasn't to validate that it was valid but ensure there were no fields we don't know how to handle.

Reply:
I'm not concerned that what they have is valid or not but that we can't reproduce their config.